### PR TITLE
Client Scripting: More chat functions

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

File: src/script/lua_api/l_client.cpp [hunk 1]
```diff
@@ -20,14 +20,38 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
+
+// display_chat_message(message)
+int ModApiClient::l_display_chat_message(lua_State *L)
+ {
+	NO_MAP_LOCK_REQUIRED;
+	
+		std::string message = luaL_checkstring(L, 1);
+		getClient(L)->displayChat(utf8_to_wide(message));
+	return 1;
+}
```
Comment (rubenwardy) [inline src/script/lua_api/l_client.cpp]: indentation errors above here

File: src/script/lua_api/l_client.cpp [hunk 2]
```diff
@@ -20,14 +20,38 @@ with this program; if not, write to the Free Software Foundation, Inc.,
...
 
 int ModApiClient::l_get_current_modname(lua_State *L)
 {
 	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
 	return 1;
 }
 
+
+// send_chat_message()
+int ModApiClient::l_send_chat_message(lua_State *L)
```
Comment (rubenwardy) [inline src/script/lua_api/l_client.cpp]: `client.say(msg)` would be better imo

Comment (nerzhul) [inline src/script/lua_api/l_client.cpp]: i agree with say but i don't like mods to send chat messages to server without user agreement. 

Cons:

* Message flooding using Lua API
* Sending messages without user agreement
* Craft bogus messages easily which can annoy other users experience

Comment (rubenwardy) [inline src/script/lua_api/l_client.cpp]: Good point

File: src/client.h [hunk 3]
```diff
@@ -557,6 +557,11 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 
 	void makeScreenshot(IrrlichtDevice *device);
 
+	void displayChat(const std::wstring &input)
```
Comment (rubenwardy) [inline src/client.h]: I suggest calling this pushToChatLog or similar
Maybe this should also be inlined and used in handlecommand_ChatMessage

**Comment by paramat** [comment]
No description and no approvals?
@nerzhul @red-001 please link to the 2nd approval, i assume nerzhul gave it one approval.

**Comment by paramat** [comment]
Lacking description.

**Pull request merged**